src/effects/Equalization.cpp

```diff
@@ -411,40 +416,121 @@ bool EffectEqualization::TransferDataToWindow(const EffectSettings &settings)
 }
 
 namespace {
-struct EqualizationTask {
-   EqualizationTask( size_t M, size_t idealBlockLen, WaveTrack &t )
-      : buffer{ idealBlockLen }
+//! Function takes an increment of samples processed, returns true to continue
+using Poller = std::function<bool(sampleCount)>;
+
+class EffectTask {
+public:
+   virtual ~EffectTask();
```

Comment:
Why not directly here `virtual ~EffectTask() = default;` ?

Reply:
Virtual functions need to have non-inline bodies compiled somewhere.  The compiler might generate them even if you make them all inline, but then it has to eliminate duplicates at link time.

If you make at least one virtual function non-inline, the compiler can follow the rule that it generates the rest of the functions only in the same translation unit and avoid the extra work of removing the duplicates.  So I do this with destructors to be a little nicer to the compiler.

Though sometimes, a template is generating the class, and then you often must inline them anyway and rely on the elimination.